src/main/java/com/groupesan/project/java/scrumsimulator/mainpackage/ui/panels/SpikeListPanel.java

```diff
@@ -0,0 +1,111 @@
+package com.groupesan.project.java.scrumsimulator.mainpackage.ui.panels;
+
+import com.groupesan.project.java.scrumsimulator.mainpackage.core.Player;
+import com.groupesan.project.java.scrumsimulator.mainpackage.core.Roles;
+import com.groupesan.project.java.scrumsimulator.mainpackage.impl.Spike;
+import com.groupesan.project.java.scrumsimulator.mainpackage.impl.SpikeStore;
+import com.groupesan.project.java.scrumsimulator.mainpackage.ui.widgets.BaseComponent;
+import com.groupesan.project.java.scrumsimulator.mainpackage.ui.widgets.BlockerWidget;
+import com.groupesan.project.java.scrumsimulator.mainpackage.ui.widgets.SpikeWidget;
+import com.groupesan.project.java.scrumsimulator.mainpackage.utils.CustomConstraints;
+import lombok.Getter;
+
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpikeListPanel extends JFrame implements BaseComponent {
+    private Player player;
+
+    @Getter
+    private List<SpikeWidget> widgets = new ArrayList<>();
+    private JPanel subPanel = new JPanel();
+
+    public SpikeListPanel(Player player) {
+        this.player = player;
+        this.init();
+    }
+
+    public void init() {
+        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        setTitle("Spike List");
+        setSize(500, 300);
+
+        JPanel myJpanel = new JPanel(new GridBagLayout());
+        myJpanel.setBorder(new EmptyBorder(10, 10, 10, 10));
+
+        subPanel.setLayout(new GridBagLayout());
+
+        JScrollPane scrollPane = new JScrollPane(subPanel);
+        scrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
+
+        refreshSpikes();
+
+        myJpanel.add(
+                scrollPane,
+                new CustomConstraints(
+                        0, 0, GridBagConstraints.WEST, 1.0, 0.8, GridBagConstraints.BOTH)); // Use BOTH to expand
+
+        JButton newSpikeButton = new JButton("New Spike");
+        if (player.getRole().getName().equals(Roles.PRODUCT_OWNER.getDisplayName())) {
+            newSpikeButton.setEnabled(false);
+        }
+        newSpikeButton.addActionListener(e -> {
+            SpikeForm form = new SpikeForm();
+            form.setVisible(true);
+            form.addWindowListener(
+                    new java.awt.event.WindowAdapter() {
+                        public void windowClosed(java.awt.event.WindowEvent windowEvent) {
+                            Spike spike = form.getSpikeObject();
+                            if (spike != null) {
+                                addSpikeWidget(spike);
+                            }
+                        }
```

Comment:
Please check this windowClosed method again, empty spike is getting added to the list when clicked on Cancel button (or) Close in the SpikeForm